large-message-core/src/main/java/com/bakdata/kafka/AbstractLargeMessageConfig.java

```diff
@@ -309,18 +303,21 @@ private Optional<AWSCredentialsProvider> getAmazonCredentialsProvider() {
         if (!isEmpty(roleArn) && !isEmpty(roleSessionName)) {
 
             if (!isEmpty(roleExternalId)) {
-                final AWSCredentialsProvider roleProvider =
-                        new STSAssumeRoleSessionCredentialsProvider.Builder(roleArn, roleSessionName)
-                                .withStsClient(AWSSecurityTokenServiceClientBuilder.defaultClient())
-                                .withExternalId(roleExternalId)
-                                .build();
+                final AwsCredentialsProvider roleProvider = StsAssumeRoleCredentialsProvider.builder()
+                        .refreshRequest(AssumeRoleRequest.builder()
+                                .roleArn(roleArn)
+                                .roleSessionName(roleSessionName)
+                                .externalId(roleExternalId)
+                                .build())
+                        .stsClient(StsClient.create())
+                        .build();
 
                 return Optional.of(roleProvider);
             }
 
             if (!isEmpty(jwtPath)) {
-                final AWSCredentialsProvider oidcProvider = WebIdentityTokenCredentialsProvider.builder()
-                        .webIdentityTokenFile(jwtPath)
+                final AwsCredentialsProvider oidcProvider = WebIdentityTokenFileCredentialsProvider.builder()
+                        .webIdentityTokenFile(new File(jwtPath).toPath())
```

Comment:
```suggestion
                        .webIdentityTokenFile(Path.of(jwtPath))
```

Reply:
Path.of is unfortunately not Java 8 compatible